tests/test_class_sh_shared_ptr_copy_move.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+
+from pybind11_tests import class_sh_shared_ptr_copy_move as m
+
+
+def test_avl_copy():
+    m.test_avl_copy()
```

Comment:
To be sure the copy and move behavior is as expected, something like:
```
    l = m.test_shp_copy()
    assert m.get_mtxt(l[0]) == "what we expect"
```
You probably have to allow for differences between platforms, using regular expressions, ideally this function from @rhaschke's still-open PR:

https://github.com/pybind/pybind11/pull/2916/files#diff-e614dc3a68da8c26363419d3c06816825b1b4344e35103c5dd004f7dcf3a6f84

```
def check_regex(expected, actual):
    result = re.match(expected + "$", actual)
    if result is None:
        pytest.fail("expected: '{}' != actual: '{}'".format(expected, actual))
```

For this you will also need the `get_mtxt` helper from test_class_sh_basic.cpp.

Reply:
Good idea,. Though, implemented as exact string compare, as in this case there should be no modification of the Foo objects after construction. The copy/move constructors are now deleted, just to ensure that.

Reply:
I have to think about this one. We don't want to leave any holes and want to know exactly what happens in all situations incl. different platforms. Maybe we need multiple `Foo` types with different combinations of (copy, move) x (ctor, assignment-op)? Is some or all of that obviously non-sensical? But let's get the CI green first with what you have right now.

Reply:
Is there anywhere where the pybind code is conditional on the class being moveable/copyable?
It may be good to add a move-only Foo and copyable Foo to test that indeed the object is not modified in any way, and it's only pointer copying.